### PR TITLE
plot: fix yaxis and grid

### PR DESCRIPTION
**Pull request by berak**
### This pullrequest changes
* make y axis point up
* fix the grid (snap to 1, 0.1, etc)
* change text rendering to show min/max values at resp. axes
* make it possible to reuse the drawing image for multiple curve plots

this hopefully fixes issues 3,4,5, and feature 4 in #961

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
Thanks but, if we revert axis directions, than for me, that kind of is unnatural. Also it will get users, who are already used to this, end up with broken code?

**Comment by berak** [comment]
@StevenPuttemans , yes, that's  true, and i'm also a bit unsure here. (let's see, if we can get more comments)

**Comment by StevenPuttemans** [comment]
We can just add thumbs up and down :+1:  :-1:

File: modules/plot/src/plot.cpp [hunk 1]
```diff
@@ -62,9 +61,17 @@ namespace cv
                 Mat _plotData = plotData.getMat();
                 //if the matrix is not Nx1 or 1xN
                 if(_plotData.cols > 1 && _plotData.rows > 1)
-                    CV_Error(Error::StsBadArg, "ERROR: Plot data must be a 1xN or Nx1 matrix.\n");
+                {
+                    std::cout << "ERROR: Plot data must be a 1xN or Nx1 matrix." << std::endl;
+                    exit(0);
```
Comment (alalek) [inline modules/plot/src/plot.cpp]: Please keep `CV_Error` here and below.

Comment (berak) [inline modules/plot/src/plot.cpp]: ah, true. no real reason to change it

**Comment by berak** [comment]
i'm somewhat not happy with this, it does not quite deliver:

* ppl will *still* get confused, what's up or down. 
* having a nice grid is nice, but if it "auto-snaps" to the data, it introduces a lot of jitter, if used real-time (like in @LaurentBerger 's histogram demo
* appending multiple plots to the same image still requires to decide on ranges beforehand, and setting them for each run, so not real usable, atm.

**Pull request closed without merging**